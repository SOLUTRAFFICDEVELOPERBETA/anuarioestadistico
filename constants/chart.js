const datalabels = {
  backgroundColor: '#fff',
  align: 'center',
  anchor: 'end',
  padding: 5,
  borderRadius: 2,
  clamp: true
}

export const BAR_OPTIONS = {
  scales: {
    xAxes: [
      {
        display: false,
        gridLines: {
          display: false
        }
      }
    ],
    yAxes: [
      {
        display: true,
        type: 'linear',
        position: 'left',
        gridLines: {
          display: false
        },
        tricks: {
          beginAtZero: true,
          precision: 0,
          fixedStepSize: 1
        }
      }
    ]
  },
  responsive: true,
  legend: {
    display: false
  },
  elements: {
    line: {
      fill: false
    }
  },
  plugins: {
    datalabels
  }
}

export const PIE_OPTIONS = {
  responsive: true,
  legend: {
    display: false
  },
  tooltips: {
    mode: 'index',
    axis: 'y'
  },
  plugins: {
    datalabels: {
      ...datalabels,
      align: 'center',
      anchor: 'center'
    }
  },
  elements: {
    line: {
      fill: false
    }
  }
}

export const LINE_OPTIONS = {
  responsive: true,
  tooltips: {
    mode: 'index',
    axis: 'y'
  },
  legend: {
    display: false
  },
  elements: {
    line: {
      fill: false
    }
  },
  plugins: {
    datalabels
  },
  scales: {
    xAxes: [
      {
        display: true,
        gridLines: {
          display: false
        }
      }
    ],
    yAxes: [
      {
        display: true,
        position: 'left',
        gridLines: {
          display: false
        },
        tricks: {
          beginAtZero: true,
          precision: 0,
          fixedStepSize: 1
        }
      }
    ]
  }
}

export const HORIZONTAL_OPTIONS = {
  responsive: true,
  tooltips: {
    mode: 'index',
    axis: 'y'
  },
  legend: {
    display: false
  },
  elements: {
    line: {
      fill: false
    }
  },
  plugins: {
    datalabels
  },
  scales: {
    xAxes: [
      {
        display: true,
        gridLines: {
          display: false
        },
        // stacked: true,
        tricks: {
          beginAtZero: true,
          precision: 0,
          fixedStepSize: 1
        }
      }
    ],
    yAxes: [
      {
        display: true,
        position: 'left',
        gridLines: {
          display: false
        },
        stacked: true
      }
    ]
  }
}

export const CHART_PIE = 'pie'
export const CHART_BAR = 'bar'
export const CHART_LINE = 'line'
export const CHART_AREA = 'area'
export const CHART_RADAR = 'radar'
export const CHART_RADIAL = 'radial'
export const CHART_COMPOSE = 'compose'

export const CHART_TYPES = {
  [CHART_LINE]: 'Linear',
  [CHART_BAR]: 'Barras',
  [CHART_AREA]: 'Area',
  [CHART_COMPOSE]: 'Compuesto',
  [CHART_PIE]: 'Pastel',
  [CHART_RADAR]: 'Radar',
  [CHART_RADIAL]: 'Radial'
}

export const CHARTS_LINES_TYPES = ['basis', 'basisClosed', 'basisOpen', 'linear', 'linearClosed', 'natural', 'monotoneX', 'monotoneY', 'monotone', 'step', 'stepBefore', 'stepAfter']

export const GRAPHIC_TYPES = ['bar', 'pie', 'line', 'doughnut', 'horizontal']

export const DEFAULT_CHART = {
  labels: ['Dato 1', 'Dato 2', 'Dato 3', 'Dato 4'],
  datasets: [
    {
      backgroundColor: ['#629749', '#ffb04c', '#ff833a', '#f05545'],
      borderColor: ['#003d00', '#bc5100', '#ac1900', '#7f0000'],
      borderWidth: 1,
      hoverBackgroundColor: ['#33691e', '#f57f17', '#e65100', '#b71c1c'],
      hoverBorderColor: ['#003d00', '#bc5100', '#ac1900', '#7f0000'],
      data: [12, 5, 4, 16],
      datalabels: {
        color: ['#629749', '#ffb04c', '#ff833a', '#f05545'],
        backgroundColor: '#fff',
        borderColor: ['#003d00', '#bc5100', '#ac1900', '#7f0000']
      }
    }
  ]
}
