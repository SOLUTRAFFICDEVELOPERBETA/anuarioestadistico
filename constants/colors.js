export const STICKY_COLOR = ['#ff7eb9', '#ff65a3', '#7afcff', '#feff9c', '#fff740'];

export const CHART_COLORS = [
    '#ea5362',
    '#54647e',
    '#5b9ee5',
    '#4bccad',
    '#9fd468',
    '#fdca61',
    '#f4a166',
    '#98ea84',
    '#7eb7e8',
    '#43463f',
    '#858adf'
];

export const COLORS = [
    '#1976d2',
    '#0d47a1',
    '#2962ff',
    '#01579b',
    '#dc004e',
    '#b71c1c',
    '#f44336',
    '#bf360c',
    '#ff9800',
    '#ff6f00',
    '#2196f3',
    '#006064',
    '#4caf50',
    '#1b5e20',
    '#546e7a',
    '#455a64',
    '#263238',
    '#212121',
    '#cccccc',
    '#ffffff'
];

export function randomHexColorCode() {
    const n = (Math.random() * 0xfffff * 1000000).toString(16);
    return '#' + n.slice(0, 6);
}
