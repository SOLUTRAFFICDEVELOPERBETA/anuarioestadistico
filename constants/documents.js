export const TITLE = 'title'
export const SUBTITLE = 'subtitle'
export const PARAGRAPH = 'paragraph'
export const DIVIDER = 'divider'
export const QUOTES = 'quotes'
export const IMAGE = 'image'
export const LIST = 'list'
export const TABLE = 'table'
export const SECTION = 'section'