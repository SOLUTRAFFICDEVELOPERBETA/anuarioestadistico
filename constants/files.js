/* eslint-disable no-unused-vars */
/* eslint-disable no-undef */
import fb from '../config/firebase';
import moment from 'moment';

/**
 * Método para subir archivos
 * @param {File} file Archivo a subir
 * @param {string} path Ruta del archivo
 * @param {string} name Nombre del archivo
 * @param {(progress) => void} onSnap Función para obtener el progreso de subida
 * @param {(error) => void} onError Función de error
 */
export function UploadFile(
    file,
    path = '/anuarioEstadistico',
    name = '',
    onSnap = (progress) => console.log('Progress', progress),
    onError = (error) => console.error('Upload Error:', error)
) {
    return new Promise((resolve, reject) => {
        const upload = new File([file], name || file.name);
        const ref = fb.storage.ref(`${path}/${upload.name}`);
        const task = ref.put(file);
        task.on(
            'state_changed',
            (snap) => {
                var pro = (snap.bytesTransferred / snap.totalBytes) * 100;
                onSnap(pro);
            },
            (error) => {
                onError(error);
            },
            () => {
                task.snapshot.ref.getDownloadURL().then((url) => {
                    resolve({
                        url,
                        path: `${path}/${ref.name}`,
                        name: ref.name
                    });
                });
            }
        );
    });
}

/**
 * Método para eliminar un archivo
 * @param {string} path Ruta del archivo
 */
export function handleDeleteFile(path) {
    return fb.storage.ref(path).delete();
}

/**
 * Método para descargar un archivo CSV o JSON
 * @param {string} data Información a descargar
 * @param {'csv' | 'json'} type Tipo de dato
 * @param {string} name Nombre del archivo
 */
export function downloadDoc(data, type, name) {
    const str = `data:text/${type};charset=utf-8,` + encodeURIComponent(data);
    const anchor = document.createElement('a');
    anchor.setAttribute('href', str);
    anchor.setAttribute('target', '_blank');
    anchor.setAttribute('download', `${name}_key_&_${moment().format('lll')}.csv`);
    document.body.appendChild(anchor);
    anchor.click();
    anchor.remove();
}
