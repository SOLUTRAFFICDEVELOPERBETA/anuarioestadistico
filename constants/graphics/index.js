/* eslint-disable no-unused-vars */
export const GENERAL_OPTIONS = {
    MIN: {
        responsive: true,
        tooltips: {
            mode: 'label'
        },
        plugins: {
            datalabels: {
                display: false
            }
        },
        legend: {
            display: false
        },
        elements: {
            line: {
                fill: false
            }
        },
        scales: {
            xAxes: [
                {
                    display: false,
                    gridLines: {
                        display: false
                    }
                }
            ],
            yAxes: [
                {
                    display: false,
                    gridLines: {
                        display: false
                    }
                }
            ]
        }
    },
    FULL: {
        responsive: true,
        tooltips: {
            mode: 'label'
        },
        legend: {
            display: true,
            position: 'bottom'
        },
        elements: {
            line: {
                fill: false
            }
        },
        scales: {
            xAxes: [
                {
                    display: true,
                    gridLines: {
                        display: false
                    }
                }
            ],
            yAxes: [
                {
                    display: true,
                    position: 'left',
                    gridLines: {
                        display: false
                    }
                }
            ]
        }
    }
};

export const POPULATION_OPTIONS = {
    MIN: {
        responsive: true,
        plugins: {
            datalabels: {
                display: false
            }
        },
        legend: {
            display: false
        },
        elements: {
            line: {
                fill: false
            }
        },
        scales: {
            xAxes: [
                {
                    display: false,
                    gridLines: {
                        display: false
                    }
                }
            ],
            yAxes: [
                {
                    display: false,
                    gridLines: {
                        display: false
                    }
                }
            ]
        },
        tooltips: {
            callbacks: {
                title: function (tooltipItem, data) {
                    return data['labels'][tooltipItem[0]['index']];
                },
                label: function (tooltipItem) {
                    return Number(tooltipItem.value).toExponential(2);
                }
            }
        }
    },
    FULL: {
        responsive: true,
        legend: {
            display: true,
            position: 'bottom'
        },
        elements: {
            line: {
                fill: false
            }
        },
        plugins: {
            datalabels: {
                display: 'auto',
                formatter: function (value, context) {
                    const num = Number(value);
                    return `${num.toExponential(1)}`;
                }
            }
        },
        tooltips: {
            callbacks: {
                title: function (tooltipItem, data) {
                    return data['labels'][tooltipItem[0]['index']];
                },
                label: function (tooltipItem) {
                    return Number(tooltipItem.value).toExponential(2);
                }
            }
        },
        scales: {
            xAxes: [
                {
                    display: true,
                    gridLines: {
                        display: false
                    },
                    ticks: {
                        callback: function (value) {
                            return Math.abs(value);
                        }
                    }
                }
            ],
            yAxes: [
                {
                    display: true,
                    position: 'left',
                    gridLines: {
                        display: false
                    },
                    ticks: {
                        callback: function (value) {
                            return parseFloat(value).toExponential(2);
                        }
                    }
                }
            ]
        }
    }
};

export const DISTRIBUTION_OPTIONS = {
    MIN: {
        responsive: true,
        plugins: {
            datalabels: {
                display: false,
                formatter: function (value, context) {
                    const num = parseFloat(value);
                    return `${num.toExponential(1)}`;
                }
            }
        },
        legend: {
            display: false
        },
        elements: {
            line: {
                fill: false
            }
        },
        tooltips: {
            callbacks: {
                title: function (tooltipItem, data) {
                    return data['labels'][tooltipItem[0]['index']];
                },
                label: function (tooltipItem) {
                    return tooltipItem.value.replace('-', '') + '%';
                }
            }
        },
        scales: {
            xAxes: [
                {
                    display: false,
                    gridLines: {
                        display: false
                    },
                    stacked: true
                }
            ],
            yAxes: [
                {
                    display: false,
                    gridLines: {
                        display: false
                    },
                    stacked: true
                }
            ]
        }
    },
    FULL: {
        responsive: true,
        legend: {
            display: true,
            position: 'bottom'
        },
        elements: {
            line: {
                fill: false
            }
        },
        plugins: {
            datalabels: {
                formatter: function (value, context) {
                    const num = Number(value);
                    return `${Math.abs(num)}%`;
                }
            }
        },
        tooltips: {
            callbacks: {
                title: function (tooltipItem, data) {
                    return data['labels'][tooltipItem[0]['index']];
                },
                label: function (tooltipItem) {
                    return tooltipItem.value.replace('-', '') + '%';
                }
            }
        },
        scales: {
            xAxes: [
                {
                    display: true,
                    gridLines: {
                        display: false
                    },
                    stacked: true,
                    labelString: 'porcentaje',
                    ticks: {
                        // Include a dollar sign in the ticks
                        callback: function (value) {
                            return Math.abs(value) + '%';
                        }
                    }
                }
            ],
            yAxes: [
                {
                    display: true,
                    position: 'left',
                    gridLines: {
                        display: false
                    },
                    stacked: true
                }
            ]
        }
    }
};

export const GRAPHIC_OPTIONS = {
    responsive: true,
    tooltips: {
        mode: 'label',
        callbacks: {
            title: function (tooltipItem, data) {
                return data['labels'][tooltipItem[0]['index']];
            },
            label: function (tooltipItem) {
                return tooltipItem.value;
            }
        }
    },
    elements: {
        line: {
            fill: false
        }
    },
    plugins: {
        datalabels: {
            display: 'auto',
            formatter: function (value, context) {
                return value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
            }
        }
    },
    scales: {
        xAxes: [
            {
                display: true,
                gridLines: {
                    display: false
                }
            }
        ],
        yAxes: [
            {
                type: 'linear',
                display: true,
                position: 'left',
                gridLines: {
                    display: false
                },
                tricks: {
                    min: 0,
                    beginAtZero: true
                }
            }
        ]
    }
};

export const DEFAULT_OPTIONS = {
    responsive: true,
    tooltips: {
        mode: 'label'
    },
    plugins: {
        datalabels: {
            display: true,
            formatter: function (value, context) {
                return value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
            }
        }
    },
    legend: {
        display: true
    },
    elements: {
        line: {
            fill: false
        }
    },
    scales: {
        xAxes: [
            {
                display: true,
                gridLines: {
                    display: false
                }
            }
        ],
        yAxes: [
            {
                display: true,
                gridLines: {
                    display: false
                },
                ticks: {
                    beginAtZero: true,
                    min: 0,
                    callback: function (value) {
                        if (!isNaN(value)) {
                            return value.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
                        }
                        return value;
                    }
                }
            }
        ]
    }
};

export const RADAR_OPTION = {
    responsive: true,
    tooltips: {
        mode: 'label'
    },
    plugins: {
        datalabels: {
            display: true
        }
    },
    legend: {
        display: true
    },
    elements: {
        line: {
            fill: false
        }
    },
    scale: {
        angleLines: {
            display: false
        },
        ticks: {
            beginAtZero: true,
            suggestedMin: 0,
            suggestedMax: 100
        }
    },
    scales: {
        xAxes: [
            {
                display: false,
                gridLines: {
                    display: false
                }
            }
        ],
        yAxes: [
            {
                display: false,
                gridLines: {
                    display: false
                }
            }
        ]
    }
};
