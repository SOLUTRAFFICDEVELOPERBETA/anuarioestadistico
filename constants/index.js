


// Módulos
import TerrestrialMobility from "../Layout/Anuario/MovilidadTerrestre"
import HistoricosGeograficos from "../Layout/Anuario/historicosGeograficos"
import Demografia from "../Layout/Anuario/demografia"
import Salud from "../Layout/Anuario/salud"
import Educacion from "../Layout/Anuario/educacion"
import Reacreacion from "../Layout/Anuario/recreacion"
import ServiciosPublicos from "../Layout/Anuario/serviciosPublicos"
import SeguridadConvivencia from "../Layout/Anuario/SeguridadConvivencia"
import ViolenciaIntrafamiliar from "../Layout/Anuario/violenciaIntrafamiliar"
import HaciendaPublica from "../Layout/Anuario/haciendaPublica"
import SectorFinanciero from "../Layout/Anuario/sectorFinanciero"
import SectorAgropecuario from "../Layout/Anuario/sectorAgropecuario"
import ConstruccionVivienda from "../Layout/Anuario/contruccionVivienda"
import SectorEmpresarial from "../Layout/Anuario/sectorEmpresarial"
import ComercioExterior from "../Layout/Anuario/comercioExterior"
import MovimientoAereo from "../Layout/Anuario/movimientoAereo"

export const PROFILE_IMG = 'https://firebasestorage.googleapis.com/v0/b/wise-365ab.appspot.com/o/assets%2Fprofile_default.jpg?alt=media&token=9779b4ee-2ee5-49c1-b24b-720e5854867b'
export const AUTH_ITEM = 'WISE_AUTH'

// Fecha
export const INPUT_DATE = 'YYYY-MM-DD'


export const COLOR_GRAPH = {
    backgroundColor: ['#629749', '#ffb04c', '#ff833a', '#f05545'],
    borderColor: ['#003d00', '#bc5100', '#ac1900', '#7f0000'],
    borderWidth: 1,
    hoverBackgroundColor: ['#33691e', '#f57f17', '#e65100', '#b71c1c'],
    hoverBorderColor: ['#003d00', '#bc5100', '#ac1900', '#7f0000'],
    datalabels: {
        color: ['#629749', '#ffb04c', '#ff833a', '#f05545'],
        backgroundColor: '#fff',
        borderColor: ['#003d00', '#bc5100', '#ac1900', '#7f0000']
    },
}

export const LinksMenu = [
    {
        label: 'Movilidad Terrestre', href: '#!',
        Component: function () {
            return <TerrestrialMobility />
        }
    },
    {
        label: 'Aspectos  Históricos  y  Geográficos', href: '#!',
        Component: function () {
            return <HistoricosGeograficos />
        }
    },
    {
        label: 'Demografía', href: '#!',
        Component: function () {
            return <Demografia />
        }
    },
    {
        label: 'Salud', href: '#!',
        Component: function () {
            return <Salud />
        }
    },
    {
        label: 'Educación', href: '#!',
        Component: function () {
            return <Educacion />
        }
    },
    {
        label: 'Recreación y zonas verdes', href: '#!',
        Component: function () {
            return <Reacreacion />
        }
    },
    {
        label: 'Servicios Públicos', href: '#!',
        Component: function () {
            return <ServiciosPublicos />
        }
    },
    {
        label: 'Seguridad y convivencia', href: '#!',
        Component: function () {
            return <SeguridadConvivencia />
        }
    },
    {
        label: 'Violencia Intrafamiliar', href: '#!',
        Component: function () {
            return <ViolenciaIntrafamiliar />
        }
    },
    {
        label: 'Hacienda Públicas', href: '#!',
        Component: function () {
            return <HaciendaPublica />
        }
    },
    {
        label: 'Sector Financiero', href: '#!',
        Component: function () {
            return <SectorFinanciero />
        }
    },
    {
        label: 'Sector Agropecuario', href: '#!',
        Component: function () {
            return <SectorAgropecuario />
        }
    },
    {
        label: 'Construcción y vivienda', href: '#!',
        Component: function () {
            return <ConstruccionVivienda />
        }
    },
    {
        label: 'Sector Empresarial', href: '#!',
        Component: function () {
            return <SectorEmpresarial />
        }
    },
    {
        label: 'Comercio Exterior', href: '#!',
        Component: function () {
            return <ComercioExterior />
        }
    },
    {
        label: 'Movimiento Aéreo', href: '#!',
        Component: function () {
            return <MovimientoAereo />
        }
    },


]


