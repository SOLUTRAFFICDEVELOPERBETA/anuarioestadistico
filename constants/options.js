// Colores de la plataforma
export const COLORS = [
    '#1976d2',
    '#0d47a1',
    '#2962ff',
    '#01579b',
    '#dc004e',
    '#b71c1c',
    '#f44336',
    '#bf360c',
    '#ff9800',
    '#ff6f00',
    '#2196f3',
    '#006064',
    '#4caf50',
    '#1b5e20',
    '#546e7a',
    '#455a64',
    '#263238',
    '#212121'
]

// Temas de la plataforma
export const THEMES = [
    {
        name: 'Por defecto',
        palette: {
            primary: {
                main: '#0d283d',
                light: '#395068',
                dark: '#000018',
                contrastText: '#ffffff'
            },
            secondary: {
                main: '#0c447c',
                light: '#496fac',
                dark: '#001e4f',
                contrastText: '#ffffff'
            },
            common: {
                main: '#a1a61f',
                light: '#d5d754',
                dark: '#6f7700',
                contrastText: '#000'
            },
            success: {
                main: '#1a8018',
                light: '#54af47',
                dark: '#005100',
                contrastText: '#fff'
            },
            error: {
                main: '#d50000',
                dark: '#9b0000',
                light: '#ff5131',
                contrastText: '#ffffff'
            },
            warning: {
                main: '#f7e92e',
                dark: '#bdb300',
                light: '#ffff5c',
                contrastText: '#fff'
            },
            info: {
                primary: '#424242',
                secondary: '#bbb',
                hint: '#f2a71b',
                disabled: '#ccc'
            }
        }
    },
    {
        name: 'Naturaleza',
        palette: {
            primary: {
                main: '#7C8C3F',
                light: '#adbc6c',
                dark: '#4e5f12',
                contrastText: '#fff'
            },
            secondary: {
                main: '#8C2323',
                light: '#c1534b',
                dark: '#590000',
                contrastText: '#fff'
            },
            common: {
                main: '#B4BF5E',
                light: '#e8f28d',
                dark: '#828f30',
                contrastText: '#000'
            },
            success: {
                main: '#1a8018',
                light: '#54af47',
                dark: '#005100',
                contrastText: '#fff'
            },
            error: {
                main: '#d50000',
                dark: '#9b0000',
                light: '#ff5131',
                contrastText: '#fff'
            },
            warning: {
                main: '#f7e92e',
                dark: '#bdb300',
                light: '#ffff5c',
                contrastText: '#fff'
            },
            info: {
                primary: '#424242',
                secondary: '#bbb',
                hint: '#B4BF5E',
                disabled: '#ccc'
            }
        }
    },
    {
        name: 'Cielo',
        palette: {
            primary: {
                main: '#F2BF27',
                light: '#fff15e',
                dark: '#bb8f00',
                contrastText: '#fff'
            },
            secondary: {
                main: '#8DD5F2',
                light: '#c0ffff',
                dark: '#5aa4bf',
                contrastText: '#fff'
            },
            common: {
                main: '#f2ab27',
                light: '#ffdd5c',
                dark: '#bb7c00',
                contrastText: '#000'
            },
            success: {
                main: '#1a8018',
                light: '#54af47',
                dark: '#005100',
                contrastText: '#fff'
            },
            error: {
                main: '#d50000',
                dark: '#9b0000',
                light: '#ff5131',
                contrastText: '#fff'
            },
            warning: {
                main: '#f7e92e',
                dark: '#bdb300',
                light: '#ffff5c',
                contrastText: '#fff'
            },
            info: {
                primary: '#40342A',
                secondary: '#bbb',
                hint: '#B4BF5E',
                disabled: '#ccc'
            }
        }
    },
    {
        name: 'Noche',
        palette: {
            primary: {
                main: '#0d283d',
                light: '#395068',
                dark: '#000018',
                contrastText: '#fff'
            },
            secondary: {
                main: '#1b4772',
                light: '#4e72a1',
                dark: '#002046',
                contrastText: '#fff'
            },
            common: {
                main: '#afbf75',
                light: '#e2f2a5',
                dark: '#7f8f48',
                contrastText: '#000'
            },
            success: {
                main: '#1a8018',
                light: '#54af47',
                dark: '#005100',
                contrastText: '#fff'
            },
            error: {
                main: '#d50000',
                dark: '#9b0000',
                light: '#ff5131',
                contrastText: '#fff'
            },
            warning: {
                main: '#f7e92e',
                dark: '#bdb300',
                light: '#ffff5c',
                contrastText: '#fff'
            },
            info: {
                primary: '#424242',
                secondary: '#bbb',
                hint: '#afbf75',
                disabled: '#ccc'
            }
        }
    },
    {
        name: 'Selva',
        palette: {
            primary: {
                main: '#208c0f',
                light: '#5bbd44',
                dark: '#005d00',
                contrastText: '#fff'
            },
            secondary: {
                main: '#D9CB0D',
                light: '#fffe53',
                dark: '#a39b00',
                contrastText: '#fff'
            },
            common: {
                main: '#89bf80',
                light: '#baf2b0',
                dark: '#5a8e53',
                contrastText: '#000'
            },
            success: {
                main: '#1a8018',
                light: '#54af47',
                dark: '#005100',
                contrastText: '#fff'
            },
            error: {
                main: '#d50000',
                dark: '#9b0000',
                light: '#ff5131',
                contrastText: '#fff'
            },
            warning: {
                main: '#f7e92e',
                dark: '#bdb300',
                light: '#ffff5c',
                contrastText: '#fff'
            },
            info: {
                primary: '#424242',
                secondary: '#bbb',
                hint: '#89bf80',
                disabled: '#ccc'
            }
        }
    },
    {
        name: 'Usuario',
        palette: {
            primary: {
                main: '#208c0f',
                light: '#5bbd44',
                dark: '#005d00',
                contrastText: '#fff'
            },
            secondary: {
                main: '#D9CB0D',
                light: '#fffe53',
                dark: '#a39b00',
                contrastText: '#fff'
            },
            common: {
                main: '#89bf80',
                light: '#baf2b0',
                dark: '#5a8e53',
                contrastText: '#000'
            },
            success: {
                main: '#1a8018',
                light: '#54af47',
                dark: '#005100',
                contrastText: '#fff'
            },
            error: {
                main: '#d50000',
                dark: '#9b0000',
                light: '#ff5131',
                contrastText: '#fff'
            },
            warning: {
                main: '#f7e92e',
                dark: '#bdb300',
                light: '#ffff5c',
                contrastText: '#fff'
            },
            info: {
                primary: '#424242',
                secondary: '#bbb',
                hint: '#89bf80',
                disabled: '#ccc'
            }
        }
    }
]
