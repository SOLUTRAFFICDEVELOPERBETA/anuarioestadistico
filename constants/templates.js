import {
    PARAGRAPH,
    TITLE,
    QUOTES,
    DIVIDER,
    LIST,
    IMAGE,
    TABLE,
    SECTION,
    CHART,
    IFRAME,
    CARD,
    SUBTITLE
} from './documents';
export const DOCUMENT_TEMPLATE = {
    title: 'Documento sin Nombre',
    fields: [
        {
            id: 'project_title',
            type: TITLE,
            value: 'Titulo'
        },
        {
            id: 'content',
            type: PARAGRAPH,
            value: 'Contenido'
        }
    ]
};

export const FIELDS = {
    [TITLE]: {
        type: TITLE,
        value: 'Titulo'
    },
    [PARAGRAPH]: {
        type: PARAGRAPH,
        value: 'Contenido'
    },
    [QUOTES]: {
        type: QUOTES,
        value: 'Cita'
    },
    [SECTION]: {
        type: SECTION,
        value: []
    },
    [CARD]: {
        type: SECTION,
        size: 6,
        value: [
            {
                id: 'card-image',
                size: 4,
                type: IMAGE,
                value: {
                    name: 'Sin nombre, haga click para cambiar',
                    path: '',
                    url:
                        'https://firebasestorage.googleapis.com/v0/b/wise-365ab.appspot.com/o/assets%2Fno-img-placeholder-min.png?alt=media&token=c8f8b86f-0bdd-4258-87a1-8b83a3fa20dd'
                }
            },
            {
                id: 'card-title',
                type: SUBTITLE,
                size: 8,
                value: 'Titulo'
            },
            {
                id: 'card-content',
                type: TITLE,
                size: 12,
                value: 'Contenido'
            }
        ]
    },
    [DIVIDER]: {
        type: DIVIDER,
        value: null
    },
    [LIST]: {
        type: LIST,
        value: []
    },
    [IMAGE]: {
        type: IMAGE,
        value: {
            name: 'Sin nombre, haga click para cambiar',
            path: '',
            url:
                'https://firebasestorage.googleapis.com/v0/b/wise-365ab.appspot.com/o/assets%2Fno-img-placeholder-min.png?alt=media&token=c8f8b86f-0bdd-4258-87a1-8b83a3fa20dd'
        }
    },
    [CHART]: {
        type: CHART,
        value: {}
    },
    [IFRAME]: {
        type: IFRAME,
        value: ''
    },
    [TABLE]: {
        type: TABLE,
        value: {
            0: [
                {
                    content: '',
                    textSize: '16',
                    bold: false,
                    italic: false,
                    cellColor: '#fff',
                    textColor: '#000'
                },
                {
                    content: '',
                    textSize: '16',
                    bold: false,
                    italic: false,
                    cellColor: '#fff',
                    textColor: '#000'
                },
                {
                    content: '',
                    textSize: '16',
                    bold: false,
                    italic: false,
                    cellColor: '#fff',
                    textColor: '#000'
                }
            ],
            1: [
                {
                    content: '',
                    textSize: '16',
                    bold: false,
                    italic: false,
                    cellColor: '#fff',
                    textColor: '#000'
                },
                {
                    content: '',
                    textSize: '16',
                    bold: false,
                    italic: false,
                    cellColor: '#fff',
                    textColor: '#000'
                },
                {
                    content: '',
                    textSize: '16',
                    bold: false,
                    italic: false,
                    cellColor: '#fff',
                    textColor: '#000'
                }
            ]
        }
    }
};

export const documents = [
    {
        id: 'WgPns4dbIvwlqUCJQ2tX ',
        title: 'Manual de uso de aplicaciones',
        uid: 'Jesus Daniel Neira',
        icon: '',
        created: 1589236235954,
        lastModified: 1589236235954,
        fields: [
            {
                id: 'project_title',
                type: TITLE,
                value: 'Manual de uso de la aplicaci贸n, [Nombre de la aplicaci贸n]'
            },
            {
                id: 'content',
                type: PARAGRAPH,
                value: 'Introducci贸n al manual'
            },
            {
                id: '70bb12',
                type: PARAGRAPH,
                value: 'Contenido sobre el manual de la aplicaci贸n'
            }
        ]
    },
    {
        id: 'Zy1Lq4eCG37U9DDZ07Qq',
        title: 'Documento en Blanco',
        uid: 'Alfredo Prado',
        icon: '',
        created: 1589236235954,
        lastModified: 1589236235954,
        fields: [
            {
                id: 'project_title'
            }
        ]
    }
];
