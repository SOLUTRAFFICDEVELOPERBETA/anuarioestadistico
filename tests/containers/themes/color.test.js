import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import ColorPalette from '../../../containers/themes/color';
import { THEMES } from '../../../constants/options';



describe('Pruebas en <ColorPalette/>', () => {
    const wrapper = shallow(
        <ColorPalette
            palette={THEMES[0].palette}
        />
    )

    test('debe mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
})