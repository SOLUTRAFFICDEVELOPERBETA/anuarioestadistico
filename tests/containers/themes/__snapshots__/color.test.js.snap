// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <ColorPalette/> debe mostrarse correctamente 1`] = `
<Fragment>
  <WithStyles(ForwardRef(IconButton))
    onClick={[Function]}
  >
    <WithStyles(ForwardRef(SvgIcon))>
      <svg
        height="24"
        width="24"
      >
        <rect
          height="24"
          style={
            Object {
              "fill": "#0d283d",
            }
          }
          width="24"
        />
      </svg>
    </WithStyles(ForwardRef(SvgIcon))>
  </WithStyles(ForwardRef(IconButton))>
  <WithStyles(ForwardRef(Popover))
    anchorEl={null}
    anchorOrigin={
      Object {
        "horizontal": "center",
        "vertical": "bottom",
      }
    }
    id="color-popover"
    onClose={[Function]}
    open={false}
    transformOrigin={
      Object {
        "horizontal": "center",
        "vertical": "top",
      }
    }
  >
    <div
      className="makeStyles-colorPalette-1"
    >
      <WithStyles(ForwardRef(SvgIcon))>
        <svg
          height="200"
          width="200"
        >
          <rect
            height="200"
            style={
              Object {
                "fill": "#0d283d",
              }
            }
            width="200"
          />
        </svg>
      </WithStyles(ForwardRef(SvgIcon))>
      <WithStyles(ForwardRef(SvgIcon))>
        <svg
          height="200"
          width="200"
        >
          <rect
            height="200"
            style={
              Object {
                "fill": "#0c447c",
              }
            }
            width="200"
          />
        </svg>
      </WithStyles(ForwardRef(SvgIcon))>
      <WithStyles(ForwardRef(SvgIcon))>
        <svg
          height="200"
          width="200"
        >
          <rect
            height="200"
            style={
              Object {
                "fill": "#a1a61f",
              }
            }
            width="200"
          />
        </svg>
      </WithStyles(ForwardRef(SvgIcon))>
      <WithStyles(ForwardRef(SvgIcon))>
        <svg
          height="200"
          width="200"
        >
          <rect
            height="200"
            style={
              Object {
                "fill": "#1a8018",
              }
            }
            width="200"
          />
        </svg>
      </WithStyles(ForwardRef(SvgIcon))>
      <WithStyles(ForwardRef(SvgIcon))>
        <svg
          height="200"
          width="200"
        >
          <rect
            height="200"
            style={
              Object {
                "fill": "#d50000",
              }
            }
            width="200"
          />
        </svg>
      </WithStyles(ForwardRef(SvgIcon))>
      <WithStyles(ForwardRef(SvgIcon))>
        <svg
          height="200"
          width="200"
        >
          <rect
            height="200"
            style={
              Object {
                "fill": "#f7e92e",
              }
            }
            width="200"
          />
        </svg>
      </WithStyles(ForwardRef(SvgIcon))>
      <WithStyles(ForwardRef(SvgIcon))>
        <svg
          height="200"
          width="200"
        >
          <rect
            height="200"
            style={
              Object {
                "fill": undefined,
              }
            }
            width="200"
          />
        </svg>
      </WithStyles(ForwardRef(SvgIcon))>
    </div>
  </WithStyles(ForwardRef(Popover))>
</Fragment>
`;
